Multiplicity.Packets/SetChestName.cs

```diff
@@ -1,94 +1,107 @@
+        public string ChestName { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SetChestName"/> class.
+        /// </summary>
+        public SetChestName()
+            : base((byte)PacketTypes.SetChestName)
+        {
+
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SetChestName"/> class.
+        /// </summary>
+        /// <param name="br">br</param>
+        public SetChestName(BinaryReader br)
+            : base(br)
+        {
+            this.ChestID = br.ReadInt16();
+            this.ChestX = br.ReadInt16();
+            this.ChestY = br.ReadInt16();
+            this.NameLength = br.ReadByte();
+            this.ChestName = String.Empty;
+
+            if (this.NameLength != 0)
+            {
+                if (this.NameLength <= 20)
+                    this.ChestName = br.ReadString();
+                else if (this.NameLength != 255)
+                    this.NameLength = 0;
+            }
+        }
+
+        public override string ToString()
+        {
+            return $"[SetChestName: ChestID = {ChestID} ChestX = {ChestX} ChestY = {ChestY} NameLength = {NameLength} ChestName = {ChestName}]";
+        }
+
+        #region implemented abstract members of TerrariaPacket
+
+        public override short GetLength()
+        {
+            const short Length = 7;
+            if (ChestName == null)
+                return Length;
+
+            return (short)(Length + ChestName.Length + 1);
+        }
+
+        public override void ToStream(Stream stream, bool includeHeader = true)
+        {
+            /*
              * Length and ID headers get written in the base packet class.
              */
-			if (includeHeader)
-			{
-				base.ToStream(stream, includeHeader);
-			}
+            if (includeHeader)
+            {
+                base.ToStream(stream, includeHeader);
+            }
 
-			/*
+            /*
              * Always make sure to not close the stream when serializing.
              * 
              * It is up to the caller to decide if the underlying stream
              * gets closed.  If this is a network stream we do not want
              * the regressions of unconditionally closing the TCP socket
              * once the payload of data has been sent to the client.
              */
-			using (BinaryWriter writer = new BinaryWriter(stream, new System.Text.UTF8Encoding(), leaveOpen: true))
-			{
-				writer.Write(ChestID);
-				writer.Write(X);
-				writer.Write(Y);
-				if (Name != null)
-				{
-					writer.Write(NameLength);
-					writer.Write(Name);
-				}
-			}
-		}
-
-		#endregion
-	}
-}
\ No newline at end of file
+            using (BinaryWriter br = new BinaryWriter(stream, new System.Text.UTF8Encoding(), leaveOpen: true))
+            {
+                br.Write(ChestID);
+                br.Write(ChestX);
+                br.Write(ChestY);
+                if (ChestName != null)
```

Comment:
If chestname is not null, but is equal to `string.Empty` should it still write out the length and name? Does Terraria expect to see bytes there even if the length is 0?